File: AiTrainerIOS/AiTrainerCore/Sources/AiTrainerCore/SettingsRepository.swift
import Foundation

public final class SettingsRepository {
    private let defaults: UserDefaults

    public init(defaults: UserDefaults = .standard) {
        self.defaults = defaults
    }

    public func loadDrawSettings() -> PracticeDrawSettings {
        guard let data = defaults.data(forKey: AppConfig.keyDrawSettings),
              let settings = try? JSONDecoder().decode(PracticeDrawSettings.self, from: data) else {
            return PracticeDrawSettings()
        }
        return settings.normalized()
    }

    public func saveDrawSettings(_ settings: PracticeDrawSettings) {
        let normalized = settings.normalized()
        if let data = try? JSONEncoder().encode(normalized) {
            defaults.set(data, forKey: AppConfig.keyDrawSettings)
        }
    }
}

File: AiTrainer/app/src/main/java/com/aitrainer/practice/data/SettingsRepository.kt
package com.aitrainer.practice.data

import android.content.Context
import com.google.gson.Gson

class SettingsRepository(context: Context) {

    private val prefs = context.applicationContext.getSharedPreferences(ProgressRepository.PREFS_NAME, Context.MODE_PRIVATE)
    private val gson = Gson()

    fun loadDrawSettings(): PracticeDrawSettings = runCatching {
        val raw = prefs.getString(KEY_DRAW_SETTINGS, null) ?: return PracticeDrawSettings()
        gson.fromJson(raw, PracticeDrawSettings::class.java).normalized()
    }.getOrDefault(PracticeDrawSettings())

    fun saveDrawSettings(settings: PracticeDrawSettings) {
        prefs.edit().putString(KEY_DRAW_SETTINGS, gson.toJson(settings.normalized())).apply()
    }

    companion object {
        private const val KEY_DRAW_SETTINGS = "ai_train_draw_settings_v1"
    }
}
